src/main/java/duke/command/FindCommand.java

```diff
@@ -0,0 +1,52 @@
+package duke.command;
+
+import duke.exception.DukeException;
+import duke.storage.Storage;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.ui.Ui;
+
+import java.util.ArrayList;
+
+/**
+ * Represents a command to find a task.
+ */
+public class FindCommand extends Command {
+    /**
+     * Term to be searched within the TaskList.
+     */
+    private String searchTerm;
+
+    /**
+     * Constructor for FindCommand.
+     *
+     * @param searchTerm The term to be searched.
+     */
+    public FindCommand(String searchTerm) {
+        this.searchTerm = searchTerm;
+    }
+
+    /**
+     * Executes the command to search for a keyword.
+     * It gets the {@literal arrayList<Task>} from within the TaskList to be able to search,
+     * then it loops through each task and searches if the description contains the keyword to be searched.
+     * It then creates a new TaskList from all the matches, and then creates a new ListCommand to list out
+     * all the tasks that contain the keyword.
+     *
```

Comment:
A shorter JavaDoc comment is sufficient, I feel your code is readable enough to understand the mechanism behind the searching of matching keywords :-)